smac/epm/rf_with_instances.py

```diff
@@ -64,20 +67,24 @@ def __init__(self, types,
         super().__init__(**kwargs)
 
         self.types = types
+        self.bounds = bounds
+        self.rng = pyrfr.regression.default_random_engine(seed)
 
-        self.rf = pyrfr.regression.binary_rss()
-        self.rf.num_trees = num_trees
-        self.rf.seed = seed
-        self.rf.do_bootstrapping = do_bootstrapping
-        self.rf.num_data_points_per_tree = n_points_per_tree
+        self.rf_opts = pyrfr.regression.forest_opts()
+        self.rf_opts.num_trees = num_trees
+        self.rf_opts.seed = seed
+        self.rf_opts.do_bootstrapping = do_bootstrapping
         max_features = 0 if ratio_features >= 1.0 else \
             max(1, int(types.shape[0] * ratio_features))
-        self.rf.max_features = max_features
-        self.rf.min_samples_to_split = min_samples_split
-        self.rf.min_samples_in_leaf = min_samples_leaf
-        self.rf.max_depth = max_depth
-        self.rf.epsilon_purity = eps_purity
-        self.rf.max_num_nodes = max_num_nodes
+        self.rf_opts.max_features = max_features
+        self.rf_opts.min_samples_to_split = min_samples_split
+        self.rf_opts.min_samples_in_leaf = min_samples_leaf
+        self.rf_opts.max_depth = max_depth
+        self.rf_opts.epsilon_purity = eps_purity
+        self.rf_opts.max_num_nodes = max_num_nodes
+
+        self.n_points_per_tree = n_points_per_tree
+        self.rf = None  # type: pyrfr.regression.binary_rss_forest
 
         # This list well be read out by save_iteration() in the solver
```

Comment:
There is a check in _train that looks if n_points_per_tree was specified to be larger than 0. If so, this value will be used in the rf_opts otherwise X.shape[0] will be used